mirgecom/diffusion.py

```diff
@@ -199,41 +233,67 @@ def diffusion_operator(discr, alpha, boundaries, u):
         if len(boundaries) != len(u):
             raise TypeError("boundaries must be the same length as u")
         return obj_array_vectorize_n_args(lambda boundaries_i, u_i:
-            diffusion_operator(discr, alpha, boundaries_i, u_i),
+            diffusion_operator(discr, quad_tag, alpha, boundaries_i, u_i),
             make_obj_array(boundaries), u)
 
     for btag, bdry in boundaries.items():
         if not isinstance(bdry, DiffusionBoundary):
             raise TypeError(f"Unrecognized boundary type for tag {btag}. "
                 "Must be an instance of DiffusionBoundary.")
 
+    actx = u.array_context
+
+    dd_quad = DOFDesc("vol", quad_tag)
+    dd_allfaces_quad = DOFDesc("all_faces", quad_tag)
+
+    if isinstance(alpha, DOFArray):
+        alpha_array = alpha
+    else:
+        alpha_array = discr.zeros(actx) + alpha
```

Comment:
How come that's necessary?

Reply:
I added that to allow specifying e.g. `alpha=2.` for the uniform case on the user side. But in retrospect I'm less sure that this is a good idea. Maybe we can talk about this today.

Reply:
As discussed, maybe rewire `project` to be able to deal with constants.

Reply:
I was able to add the constant case to `project` easily enough, but to get things working I needed to add special handling for a couple of other operations too (see code [here](https://github.com/majosm/mirgecom/commit/a493803d8e48b8074980c71d48e682e9c99a2ee0#diff-8c80087b48245eecc882a14fa78e26b0891982bff838e184fcc80bbb396b7465R45-R56)). I'm on the fence about this... What do you think?

Reply:
I actually like these modifications just fine.